static/src/javascripts/projects/common/modules/experiments/ab-core.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import {
+    getMvtNumValues,
+    getMvtValue,
+} from 'common/modules/analytics/mvt-cookie';
+import config from 'lib/config';
+import { isExpired } from 'lib/time-utils';
+import { getVariantFromLocalStorage } from './ab-local-storage';
+import { getVariantFromUrl } from './ab-url';
+import { NOT_IN_TEST } from './ab-constants';
+import { isTestSwitchedOn } from './ab-utils';
+
+// We only take account of a variant's canRun function if it's defined.
+// If it's not, assume the variant can be run.
+const variantCanBeRun = (variant: Variant): boolean =>
+    !(variant.canRun && !variant.canRun()) && variant.id !== NOT_IN_TEST;
+
+const testCanBeRun = (test: ABTest): boolean => {
+    const expired = isExpired(test.expiry);
+    const isSensitive = config.page.isSensitive;
+    const shouldShowForSensitive = !!test.showForSensitive;
+    const isTestOn = isTestSwitchedOn(test.id);
+    const canTestBeRun = !test.canRun || test.canRun();
+
+    return (
+        (isSensitive ? shouldShowForSensitive : true) &&
+        isTestOn &&
+        !expired &&
+        canTestBeRun
+    );
+};
+
+// Determine whether the user is in the test or not and return the associated
+// variant ID, based on the MVT cookie segmentation.
+//
+// The test population is just a subset of MVT ids. A test population must
+// begin from a specific value. Overlapping test ranges are permitted.
+const computeVariantFromMvtCookie = (test: ABTest): ?Variant => {
```

Comment:
This is pretty much the only bit of the old A/B testing code that I've preserved.

In terms of file structure, we now have:
- `ab-tests`: entry point, depends on all individual test files
- `ab-core`: the essential logic for determining which variants will run on  a given pageview
- `ab-local-storage`: reading/writing to localStorage
- `ab-url`: reading from the URL
- `ab-utils`: utils for transforming A/B test data structures.